Automattic/jetpack
# Avoid using user/session info for nonce verification

When a user is logged in to the site the nonce verification fails. This is because the nonce is created using the user data, and verified using user data which might not be available. This is due to the way the comment form is posted. 

Fixes https://github.com/Automattic/jetpack/issues/32552

Related work
D156337-code

## Proposed changes:

* To fix the nonce verification errors, the way we create and verify the nonce has been changed now to rely on user data.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* In incognito window login to wordpress.com and wp-admin of the site
* Verify that the original issue is reproducible on your site
* Sync this PR to an atomic site using rsync or the beta plugin 
* Navigate to a post and try to post a comment
* Verify that the comment is posted without any nonce verification errors
